### PR TITLE
GUI: Fix displaying dynamic engine options widgets in the Edit Game dialog

### PR DESCRIPTION
**Pull request by ccawley2011**
This fixes [Trac #12027](https://bugs.scummvm.org/ticket/12027).

### CODE REVIEW

File: gui/editgamedialog.cpp [hunk 1]
```diff
@@ -192,7 +192,10 @@ EditGameDialog::EditGameDialog(const String &domain)
 
 		const MetaEngineDetection &metaEngineDetection = metaEnginePlugin->get<MetaEngineDetection>();
 		metaEngineDetection.registerDefaultSettings(_domain);
-		_engineOptions = metaEngineDetection.buildEngineOptionsWidgetStatic(tab, "GameOptions_Engine.Container", _domain);
+		if (enginePlugin)
+			_engineOptions = enginePlugin->get<MetaEngine>().buildEngineOptionsWidgetDynamic(tab, "GameOptions_Engine.Container", _domain);
+		if (!_engineOptions)
+			_engineOptions = metaEngineDetection.buildEngineOptionsWidgetStatic(tab, "GameOptions_Engine.Container", _domain);
```
Comment (bluegr) [inline gui/editgamedialog.cpp:199]: These two are mutually exclusive, right? Thus, the second if statement should be:

    else if (!_engineOptions)

**Comment by criezy** [comment]
I am wondering if the best solution is what you have done in this pull request, or if it would be better to instead implement `buildEngineOptionsWidgetStatic` in `MohawkMetaEngineDetection` (that would mean duplicating some code, but it would be only a few lines of code).

Two questions that I have are:
 - whether we might want for some engines to have a separate widget for the launcher Edit game dialog, and the in-game options dialog and thus implement both `buildEngineOptionsWidgetStatic` and `buildEngineOptionsWidgetDynamic` with a different implementation. 
 - if in some cases we might want an in-game options widget but nothing in the edit game dialog.

If the answer to both is no, then what you are proposing here looks fine. Although we would need to also update the documentation for both `buildEngineOptionsWidgetStatic` and `buildEngineOptionsWidgetDynamic` as the current documentation indicates that the former is used for the Edit game dialog, and the latter is used only for the in-game options dialog.

If the answer to the first question is yes, but the answer to the second question is no, then we would only need a small tweak to this pull request (try `buildEngineOptionsWidgetStatic` first).

But if the answer to the second question is yes, then we might need to keep the `EditGameDialog` code as it is, and instead implement `buildEngineOptionsWidgetStatic` in `MohawkMetaEngineDetection`.

Note: There is also a small difference in behaviour if the engine is not enabled, as then `buildEngineOptionsWidgetDynamic` cannot be used while `buildEngineOptionsWidgetStatic` can still be used. This means that with this pull request, if we disable the mohawk engine, the engine tab will not be present in the Edit game dialog. This probably doesn't matter though since we would not be able to start the game anyway.

**Comment by bluegr** [comment]
In my opinion, the engine options that are available in the edit game dialog and the in-game options widget should *ideally* be the same. However, there are cases where changing an option while a game is running would need a restart of the game.

Thus, I am in favor of what has been done in this pull request to fix the current issue with the game options, but ideally it would be preferable to have separate options when editing a game entry from the ones that are available in-game.

So, I too agree that more work is needed for this feature, although we can merge this pull request as an intermediate solution and future work on this can be done in a separate pull request.

**Comment by ccawley2011** [comment]
The options dialog for Myst already uses `ConfMan.getActiveDomainName()` to show or hide options based on where it appears, and the `kSupportsChangingOptionsDuringRuntime` engine option exists to control whether or not the options dialog appears in the in-game options dialog. As such, I can't see any scenario where it would be useful to have two completely separate dialogs, especially since it's not possible to directly link between the detection plugin and the engine plugin.

**Comment by OMGPizzaGuy** [comment]
Fixes runtime options for ULTIMA8 - Thanks!

**Comment by bluegr** [comment]
Thanks! This fixes the issue in 3 engines. Merging

**Pull request merged**